spec/models/project_mediaflux_spec.rb

```diff
@@ -1,80 +1,92 @@
-        end
-
         it "raises an error" do
+          # Make the project once
+          described_class.create!(project: project, session_id: current_user.mediaflux_session)
+          # It should raise a duplicate namespace error the second time
           expect do
-            described_class.create!(project: project, session_id: "test-session-token")
-          end.to raise_error(StandardError) #Raises standard error when a duplicate project already exists
+            described_class.create!(project: project, session_id: current_user.mediaflux_session)
+          end.to raise_error(MediafluxDuplicateNamespaceError) # Raises custom error when a duplicate project already exists
         end
       end
 
       context "when the parent colllection does not exist" do
-        let(:parent_metadata_request) { instance_double(Mediaflux::Http::AssetMetadataRequest, metadata: {}, "error?": true) }
-        let(:parent_collection_request) { instance_double(Mediaflux::Http::AssetCreateRequest, id: 567, "error?": false) }
+        let(:mediaflux_id) { described_class.create!(
+                              project: project, 
+                              session_id: current_user.mediaflux_session) }
+        let(:mediaflux_metadata) do
+           Mediaflux::Http::AssetMetadataRequest.new(
+                              session_token: current_user.mediaflux_session, 
+                              id: mediaflux_id
+                              ).metadata
+        end
+        let(:project_parent) { Rails.configuration.mediaflux["api_root_collection"] }
 
         before do
-          allow(Mediaflux::Http::AssetCreateRequest).to receive(:new).with(hash_including(session_token: "test-session-token", name: "tigerdata",
-                                                                                          namespace: "/td-test-001")).and_return(parent_collection_request)
+          # Destroy the parent collection
+          Mediaflux::Http::NamespaceDestroyRequest.new(
+            session_token: current_user.mediaflux_session,
+            namespace: project_parent
+          ).destroy
         end
 
-        it "creates a project namespace and collection and parent collection and returns the mediaflux id" do
-          mediaflux_id = described_class.create!(project: project, session_id: "test-session-token")
-          expect(namespace_request).to have_received("error?")
-          expect(parent_collection_request).to have_received("error?")
-          expect(asset_create_request).to have_received("id")
-          expect(mediaflux_id).to be "123"
+        it "creates a project namespace" do
+          namespace_path = mediaflux_metadata[:namespace]
+          namespace_metadata = Mediaflux::Http::NamespaceDescribeRequest.new(
+            session_token: current_user.mediaflux_session, 
+            path: namespace_path
+            ).metadata
+          
+          expect(namespace_metadata[:description]).to match(/Namespace for/)
+        end
```

Comment:
@bess since this is now talking to mediflux, could we move all these tests into 1 it block? Aka 3 expects in the test instead of 3 tests?